superset/migrations/shared/migrate_viz/base.py

```diff
@@ -156,9 +156,7 @@ def downgrade_slice(cls, slc: Slice) -> Slice:
         return slc
 
     @classmethod
-    def upgrade(cls) -> None:
-        bind = op.get_bind()
-        session = db.Session(bind=bind)
+    def upgrade(cls, session: Session) -> None:
         slices = session.query(Slice).filter(Slice.viz_type == cls.source_viz_type)
```

Comment:
Can this not be `db.session` for all occurrences, i.e., CLI and Alembic migration?

Reply:
I tried using `db.session` but it completely broke CI. I decided to remove session creation from the base class and leave that responsibility with the migration callers. This preserves the current behavior of Alembic migrations and at the same time allows the migration to be called from commands.